imcache_test.go

```diff
@@ -941,6 +942,53 @@ func TestImcache_GetAll_SlidingExpiration(t *testing.T) {
 	}
 }
 
+//GetAndRemove Tests Init
+func TestImcache_GetAndRemove(t *testing.T) {
```

Comment:
Could you also add a test to confirm the eviction callback is called with proper args? Example: https://github.com/erni27/imcache/blob/master/imcache_test.go#L1461 (we want to cover both cases, if the entry is expired and removed).